exercises/say/tests/say.rs

```diff
@@ -0,0 +1,153 @@
+// as specified in the exercise description.  
+
+// As an extra exercise, the last test checks say::encode_u64(u64) String
+// parsing a max value u64.  This is more complex of a task, but this method
+// can be wrapped to provide say::encode functionality with checks.
+
+
+#[test]
+fn test_zero() {
+    assert_eq!(say::encode(0), Ok(String::from("zero")));
+}
+
+#[test]
+fn test_one() {
+   assert_eq!(say::encode(1), Ok(String::from("one")));
+}
+
+#[test]
+#[ignore]
+fn test_fourteen() {
+   assert_eq!(say::encode(14), Ok(String::from("fourteen")));
+}
+
+#[test]
+#[ignore]
+fn test_twenty() {
+   assert_eq!(say::encode(20), Ok(String::from("twenty")));
+}
+
+#[test]
+#[ignore]
+fn test_twenty_two() {
+   assert_eq!(say::encode(22), Ok(String::from("twenty-two")));
+}
+
+#[test]
+#[ignore]
+fn test_one_hundred() {
+   assert_eq!(say::encode(100), Ok(String::from("one hundred")));
+}
+
+// note, using American style with no and
+#[test]
+#[ignore]
+fn test_one_hundred_twenty() {
+   assert_eq!(say::encode(120), Ok(String::from("one hundred twenty")));
+}
+
+#[test]
+#[ignore]
+fn test_one_hundred_twenty_three() {
+   assert_eq!(say::encode(123), Ok(String::from("one hundred twenty-three")));
+}
+
+#[test]
+#[ignore]
+fn test_one_thousand() {
+   assert_eq!(say::encode(1000), Ok(String::from("one thousand")));
+}
+
+#[test]
+#[ignore]
+fn test_one_thousand_two_hundred_thirty_four() {
+   assert_eq!(say::encode(1234), Ok(String::from("one thousand two hundred thirty-four")));
+}
+
+// note, using American style with no and
+#[test]
+#[ignore]
+fn test_eight_hundred_and_ten_thousand() {
+   assert_eq!(say::encode(810_000), Ok(String::from("eight hundred ten thousand")));
+}
+
+#[test]
+#[ignore]
+fn test_one_million() {
+   assert_eq!(say::encode(1_000_000), Ok(String::from("one million")));
+}
+
+// note, using American style with no and
+#[test]
+#[ignore]
+fn test_one_million_two() {
+   assert_eq!(say::encode(1_000_002), Ok(String::from("one million two")));
+}
+
+#[test]
+#[ignore]
+fn test_1002345() {
+   assert_eq!(
+            say::encode(1_002_345),
+            Ok(String::from("one million two thousand three hundred forty-five")));
+}
+
+#[test]
+#[ignore]
+fn test_one_billion() {
+   assert_eq!(say::encode(1_000_000_000), Ok(String::from("one billion")));
+}
+
+#[test]
+#[ignore]
+fn test_987654321123() {
+   assert_eq!(say::encode(987_654_321_123), 
+                Ok(String::from("nine hundred eighty-seven billion \
+                 six hundred fifty-four million \
+                 three hundred twenty-one thousand \
+                 one hundred twenty-three")));
+}
+
+/*
+//  Testing bounds given in problem, which requires a signed data type.
+//
+//  How could data type selection use Rust compiler to eliminate
+//  issues with receiving negative number at compile time, 
+//  rather than returning error?
+*/
+#[test]
+#[ignore]
+fn test_negative() {
+    assert!(if let Err(_) = say::encode(-1) { true } else { false });
```

Comment:
Sounds good.  I struggled with needs of exercise vs needs of language.  However, the goal is to teach the language.  And I agree that this means using it for the best conditions.